src/output.c

```diff
@@ -140,13 +173,15 @@ static void *outputThread(void *data) {
    rsFrameDestroy(&frame);
    rsDecompressDestroy(&decompress);
    rsBufferDestroy(&output->frames);
+   rsAudioEncoderDestroy(&audioenc);
 
    outputCommand(output->config->postOutputCommand);
    return NULL;
 }
 
-void rsOutputCreate(RSOutput *output, const RSConfig *config) {
+void rsOutputCreate(RSOutput *output, const RSConfig *config, RSAudio *audio) {
    output->config = config;
+   output->audio = audio;
```

Comment:
We should probably make a copy of the output data similar to `rsBufferCircleExtract`

Reply:
RSAudioEncoder receives the copy of data. In fact, the mainloop could create the RSAduioEncoder object and pass that to output, because right now output gets the RSAudio object only to create the RSAudioEncoder. Will change that

Reply:
It would probably be cleaner (imo) to create a copy of the data and pass it into output which gives it to the encoder frame by frame.

Reply:
Since RSAudio already creates the encoder (sets the encoders parameters such as bitrate, sample rate, input/output format etc.) it makes sense to copy all the samples there and have the whole "create encoder" in one place. 
Right now Output gets a pointer to RSAudio so that it can create the RSAudioEncoder. Just creating the encoder copies the collected samples, and rightfully so imo, it has to do that wrap around in the circle_static. As is now output doesn't even have to include circle_static. It only creates the encoder, calls EncodeFrame in a loop and destroys the encoder.